python/iceberg/core/filesystem/s3_filesystem_wrapper.py

```diff
@@ -0,0 +1,283 @@
+
+    expiration = STS_CACHED_RESPONSE[role_arn].get("Credentials", dict()).get("Expiration")
+    curr_dt = datetime.datetime.now(tz=datetime.timezone.utc)
+
+    return (expiration - curr_dt).total_seconds() < REFRESH_THRESHOLD
+
+
+class S3FileSystemWrapper(FileSystem):
+    _FS = dict()
+
+    def __init__(self, conf=None, anon=False):
+        self.anon = anon
+        self.conf = conf
+        self.iam_role_arn = "default"
+
+        if conf is not None:
+            self.iam_role_arn = conf.get("hive.aws_iam_role", "default")
+
+    def open(self, path, mode='rb'):
+        if mode == "rb":
+            return S3File(path, self.iam_role_arn)
+
+        elif mode == "w":
+            s3fs_obj = S3FileSystem()
+            return s3fs_obj.open(S3FileSystemWrapper.fix_s3_path(path), mode=mode)
+
+    def delete(self, path):
+        raise NotImplementedError()
+
+    def stat(self, path):
+        is_dir = False
+        st = dict()
+        try:
+            st = S3FileSystemWrapper.info(S3FileSystemWrapper.fix_s3_path(path), refresh=True)
+        except RuntimeError:
+            # must be a directory or subsequent du will fail
+            du = S3FileSystemWrapper.du(S3FileSystemWrapper.fix_s3_path(path), refresh=True)
+            if len(du) > 0:
+                is_dir = True
+            length = 0
+            for key, val in du.items():
+                length += val
+
+        return FileStatus(path=path, length=st.get("Size", length), is_dir=is_dir,
+                          blocksize=None, modification_time=st.get("LastModified"), access_time=None,
+                          permission=None, owner=None, group=None)
+
+    @staticmethod
+    def full_file(url, role_arn="default", use_requests=True):
+        if use_requests:
+            url = S3FileSystemWrapper.get_presigned_urls([url])[0]
+            stream = requests.get(url, stream=True).content
+        else:
+            stream = S3File(url, role_arn).read()
+
+        return io.BytesIO(stream)
+
+    @staticmethod
+    def get_presigned_urls(urls, role_arn="default"):
+        signed_urls = []
+        for url in urls:
+            parsed_url = urlparse(S3FileSystemWrapper.fix_s3_path(url))
+            signed_urls.append(get_s3(role_arn, "client").generate_presigned_url(ClientMethod='get_object',
+                                                                                 Params={'Bucket': parsed_url.netloc,
+                                                                                         'Key': parsed_url.path[1:]}))
+        return signed_urls
+
+    @staticmethod
+    def du(url):
+        raise NotImplementedError()
+
+    @staticmethod
+    def info(url):
+        raise NotImplementedError()
+
+    @staticmethod
+    def fix_s3_path(path):
+        if path.startswith("s3n"):
+            path = path.replace("s3n://", "s3://")
+        return path
+
+
+class S3File(object):
+    MAX_CHUNK_SIZE = 4 * 1048576
+    MIN_CHUNK_SIZE = 2 * 65536
```

Comment:
Parquet reader grabs 64k chunks of data.  Performance was better for larger data pulls, so that more of these hit the already read-in buffer. 4mb seemed to get the most benefit without reading too much.   On some small columns though, the whole column chunk is pretty small for the row-group and it did better when this was set to something closer to the size of the column chunk. 

Right now, the chunk_size it's getting set in the parquet reader if the file object that's passed in has the buffer_remote_reads attribute.  There's probably some better ways of going about this, I'd love to hear your thoughts.